resource/index_bench_test.go

```diff
@@ -9,12 +9,48 @@ import (
 
 func BenchmarkGetResource(b *testing.B) {
 	index := NewIndex()
-	for i := 0; i < 1000; i++ {
-		index.Bind(fmt.Sprintf("route%d", i), schema.Schema{}, nil, DefaultConf)
+	for i := 1; i <= 1000; i++ {
+		index.Bind(fmt.Sprintf("route%04d", i), schema.Schema{}, nil, DefaultConf)
 	}
-	b.ResetTimer()
-	b.ReportAllocs()
-	for i := 0; i < b.N; i++ {
-		index.GetResource("route80", nil)
+	tests := map[string]string{
+		"1st":      "route0001",
+		"100th":    "route0100",
+		"1000th":   "route1000",
+		"NotFound": "notfound",
 	}
+	for name, rsrc := range tests {
+		b.Run(name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				index.GetResource(rsrc, nil)
+			}
+		})
+	}
+}
+
+func BenchmarkGetResourceDepth(b *testing.B) {
+	r := NewIndex()
+	foo := r.Bind("foo", schema.Schema{}, nil, DefaultConf)
+	bar := foo.Bind("bar", "f", schema.Schema{Fields: schema.Fields{"f": {}}}, nil, DefaultConf)
+	bar.Bind("baz", "f", schema.Schema{Fields: schema.Fields{"f": {}}}, nil, DefaultConf)
+	for _, p := range []string{"foo", "foo.bar", "foo.bar.baz"} {
+		_, found := r.GetResource("foo", nil)
+		if !found {
+			b.Errorf("path %q cannot be found", p)
+		}
+	}
+	b.Run("OneLevel", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_, _ = r.GetResource("foo", nil)
+		}
+	})
+	b.Run("TwoLevels", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_, _ = r.GetResource("foo.bar", nil)
+		}
+	})
+	b.Run("ThreeLevels", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_, _ = r.GetResource("foo.bar.baz", nil)
+		}
+	})
```

Comment:
I guess you could use a `map [string]string` and for loop here as well, making the code more DRY.